src/anomaly/metaml.rs

```diff
@@ -1,214 +1,267 @@
+    /// #     pub fn new() -> Self {
+    /// #         Self {}
+    /// #     }
+    /// # }
+    /// # impl MetaMLModel for ImplMetaMLModel {
+    /// #     fn train(&mut self, dataset: MetaMLDataset) { }
+    /// #     fn predict(&self, features: &[f32; 6]) -> f32 { unimplemented!(); }
+    /// #     fn load(path: &Path) -> Result<Self, String> { unimplemented!(); }
+    /// #     fn save(&self, path: &Path) -> Result<(), String> { unimplemented!(); }
+    /// # }
+    /// #
+    /// # let features: Vec<[f32; 6]> = vec![[0.0, 0.0, 0.0, 0.0, 0.0, 0.0]];
+    /// # let targets: Vec<f32> = vec![0.0];
+    /// let mut model: ImplMetaMLModel = ImplMetaMLModel::new();
+    /// let dataset: MetaMLDataset = MetaMLDataset::new(features, targets).unwrap();
+    ///
+    /// // Train the model on the dataset
+    /// model.train(dataset);
+    ///
+    /// // Do something with your trained model
+    /// ```
+    fn train(&mut self, dataset: MetaMLDataset);
+
+    /// Makes a prediction given a trained model and 6 feature values
+    ///
+    /// # Panics
+    /// * If the model hasn't been trained
+    ///
+    /// # Examples
+    /// ```
+    /// # use clam::anomaly::metaml::{MetaMLDataset, MetaMLModel};
+    /// # use ndarray::{Array1, Array2, arr1, arr2};
+    /// # use std::path::Path;
+    /// #
+    /// # struct ImplMetaMLModel {};
+    /// # impl ImplMetaMLModel {
+    /// #     pub fn new() -> Self {
+    /// #         Self {}
+    /// #     }
+    /// # }
+    /// # impl MetaMLModel for ImplMetaMLModel {
+    /// #     fn train(&mut self, dataset: MetaMLDataset) { }
+    /// #     fn predict(&self, features: &[f32; 6]) -> f32 { 0.0 }
+    /// #     fn load(path: &Path) -> Result<Self, String> { unimplemented!(); }
+    /// #     fn save(&self, path: &Path) -> Result<(), String> { unimplemented!(); }
+    /// # }
+    /// #
+    /// # let features: Vec<[f32; 6]> = vec![[0.0, 0.0, 0.0, 0.0, 0.0, 0.0]];
+    /// # let targets: Vec<f32> = vec![0.0];
+    /// let mut model: ImplMetaMLModel = ImplMetaMLModel::new();
+    /// # let dataset: MetaMLDataset = MetaMLDataset::new(features, targets).unwrap();
+    ///
+    /// // Model must be trained before using .predict()
+    /// model.train(dataset);
+    ///
+    /// // Input data for the model to make a prediction with
+    /// // This is just random example data
+    /// let input_data: &[f32; 6] = &[4.028, 5.758, 1.402, 0.927, 0.005, 5.502];
+    ///
+    /// // Make a prediction with your model from the data
+    /// let prediction: f32 = model.predict(input_data);
+    ///
+    /// // Do something with the prediction
+    /// println!("The model predicted {prediction:.4}");
+    /// ```
+    fn predict(&self, features: &[f32; 6]) -> f32;
+
+    /// Loads a trained meta-ml model from disk.
+    ///
+    /// # Panics
+    /// * If the serialized model cannot be read from the input file path
+    /// * If the trained model cannot be deserialized
+    ///
+    /// # Examples
+    /// ```no_run
```

Comment:
Nix the no_run or add a comment about why this cannot be run, ie "I cannot get x to work because of y, so this is disabled for now"